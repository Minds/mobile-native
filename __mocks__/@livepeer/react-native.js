// mock @livepeer/react-native
export const LivepeerConfig = {
  setConfig: jest.fn(),
  getConfig: jest.fn(),
  get: jest.fn(),
  set: jest.fn(),
};

export const createReactClient = jest.fn();
export const createLivepeerClient = jest.fn();
export const Player = jest.fn();
