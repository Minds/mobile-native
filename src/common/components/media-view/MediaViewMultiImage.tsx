import { ImageSource } from 'expo-image';
import React, { useMemo } from 'react';
import {
  ImageURISource,
  TouchableOpacity,
  TouchableOpacityProps,
} from 'react-native';
import type ActivityModel from '~/newsfeed/ActivityModel';

import { Column, Row, Spacer } from '../../ui';
import SmartImage, { SmartImageProps } from '../SmartImage';
import { IS_TENANT } from '~/config/Config';
import sp from '~/services/serviceProvider';

const FIXED_HEIGHT = 220;
const BORDER_RADIUS = 3;

type PropsType = {
  fullWidth?: boolean;
  entity: ActivityModel;
  style?: any;
  ignoreDataSaver?: boolean;
  mode?: ImageProps['contentFit'];
  /**
   * @param {number} index - the index of the image
   */
  onImagePress: (index: number) => void;
  onImageLongPress: (imageSource: ImageURISource) => void;
};

type ImageProps = SmartImageProps & { source: ImageSource };

/**
 * Used for displaying multi images
 */
export default function MediaViewMultiImage({
  entity,
  ignoreDataSaver,
  onImagePress,
  onImageLongPress,
}: PropsType) {
  let images: ImageProps[] = useMemo(
    () =>
      entity.custom_data.map((image, index) => {
        if (IS_TENANT) {
          const source = entity.site_membership
            ? image.blurhash
            : {
                uri: image.src,
                headers: sp.api.buildHeaders(),
              };
          return {
            source,
            onPress: () => onImagePress(index),
            onLongPress: () => onImageLongPress(source),
            ignoreDataSaver,
            blurhash: entity.site_membership ? undefined : image.blurhash,
          };
        }
        const source = {
          uri: image.src,
          headers: sp.api.buildHeaders(),
        };
        return {
          source,
          locked: entity?.isLocked(),
          blurred: entity?.shouldBeBlured(),
          onPress: () => onImagePress(index),
          onLongPress: () => onImageLongPress(source),
          ignoreDataSaver,
          blurhash: image.blurhash,
        };
      }),
    [entity, ignoreDataSaver, onImageLongPress, onImagePress],
  );

  return (
    <Row top="M" horizontal="M" containerStyle={styles.container}>
      <Column flex right="XS">
        <ImageItem {...(images[0] as any)} />
        {images.length === 4 && (
          <>
            <Spacer top="S" />
            <ImageItem {...(images[2] as any)} />
          </>
        )}
      </Column>
      <Column flex left="XS">
        <ImageItem {...(images[1] as any)} />
        {images.length > 2 && (
          <>
            <Spacer top="S" />
            <ImageItem {...(images[images.length - 1] as any)} />
          </>
        )}
      </Column>
    </Row>
  );
}

const ImageItem = ({
  onPress,
  onLongPress,
  ...smartImageProps
}: TouchableOpacityProps & ImageProps) => (
  <TouchableOpacity
    onPress={onPress}
    onLongPress={onLongPress}
    style={styles.image}
    activeOpacity={1}
    testID={'image-' + smartImageProps.source?.uri}>
    <SmartImage {...smartImageProps} contentFit="cover" style={styles.image} />
  </TouchableOpacity>
);

const styles = sp.styles.create({
  container: {
    height: FIXED_HEIGHT,
  },
  image: {
    flex: 1,
    height: '100%',
    borderRadius: BORDER_RADIUS,
    shadowOffset: { width: 0, height: 1 },
    shadowColor: 'rgba(0, 0, 0, 0.4)',
    shadowOpacity: 0.4,
    shadowRadius: 0,
    elevation: 2,
  },
});
