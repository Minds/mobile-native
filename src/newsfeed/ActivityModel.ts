import { runInAction, action, observable, decorate } from 'mobx';
import FastImage from 'react-native-fast-image';
import { FlatList, Alert, Platform } from 'react-native';

import BaseModel from '../common/BaseModel';
import UserModel from '../channel/UserModel';
import wireService from '../wire/WireService';
import sessionService from '../common/services/session.service';
import {
  setPinPost,
  deleteItem,
  unfollow,
  follow,
} from '../newsfeed/NewsfeedService';
import api from '../common/services/api.service';

import { GOOGLE_PLAY_STORE, MINDS_CDN_URI, MINDS_URI } from '../config/Config';
import i18n from '../common/services/i18n.service';
import logService from '../common/services/log.service';
import type { ThumbSize, LockType } from '../types/Common';
import type GroupModel from '../groups/GroupModel';
import { SupportTiersType } from '../wire/WireTypes';
import mindsService from '../common/services/minds.service';
import NavigationService from '../navigation/NavigationService';
import { showNotification } from '../../AppMessages';

type Thumbs = Record<ThumbSize, string> | Record<ThumbSize, string>[];

/**
 * Activity model
 */
export default class ActivityModel extends BaseModel {
  // Observable properties
  @observable pinned: boolean = false;
  @observable time_created: string = '';
  @observable message: string = '';
  @observable title: string = '';
  @observable mature: boolean = false;
  @observable edited: '0' | '1' = '0';
  @observable paywall: true | '1' | '' = '';

  // decorated observables
  'is:following': boolean;
  'thumbs:down:count': number;
  'thumbs:up:count': number;
  'comments:count': number;
  'thumbs:down:user_guids': Array<number>;
  'thumbs:up:user_guids': Array<number>;
  seen?: boolean;
  rowKey?: string;
  description?: string; // on image objects in some cases the message is on description field
  containerObj?: GroupModel;
  remind_object?: ActivityModel;
  ownerObj!: UserModel;
  listRef?: FlatList<any>;
  thumbnails?: Thumbs;
  paywall_unlocked: boolean = false;
  guid: string = '';
  subtype: string = '';
  entity_guid: string = '';
  owner_guid: string = '';
  custom_type?: string;
  custom_data?: Array<any> | any;
  nsfw?: Array<number>;
  flags?: any;
  reminds: number = 0;
  impressions: number = 0;
  perma_url?: string;
  cinemr_guid?: string;
  thumbnail_src?: string;
  dontPin?: boolean;
  boosted?: boolean;
  wire_threshold?: {
    type: 'token' | 'money';
    min: number;
    support_tier: SupportTiersType;
  } | null;
  _preview?: boolean;
  attachments?: {
    attachment_guid: string;
  };
  type?: string;
  permaweb_id?: string;
  remind_deleted?: boolean;
  remind_users?: Array<UserModel>;

  /**
   * Mature visibility flag
   */
  @observable mature_visibility: boolean = false;

  /**
   * Is visible in flat list
   */
  @observable is_visible: boolean = false;

  /**
   *  List reference setter
   */
  set _list(value) {
    this.__list = value;

    // the reminded object need access to the metadata service too
    if (this.remind_object) {
      this.remind_object._list = value;
    }
  }

  /**
   *  List reference getter
   */
  get _list() {
    return this.__list;
  }

  /**
   * Block owner
   */
  async blockOwner() {
    await super.blockOwner();
    if (this._list) {
      this._list.refresh();
    }
  }

  /**
   * Unblock owner
   */
  async unblockOwner() {
    await super.unblockOwner();
    if (this._list) {
      this._list.refresh();
    }
  }

  /**
   * Has media
   */
  hasMedia(): boolean {
    const type = this.custom_type || this.subtype;
    switch (type) {
      case 'image':
      case 'video':
      case 'batch':
        return true;
    }
    if (this.perma_url) {
      return true;
    }
    return false;
  }

  /**
   * Has an attached video
   */
  hasVideo(): boolean {
    return (this.custom_type || this.subtype) === 'video';
  }

  /**
   * Has an attached image
   */
  hasImage(): boolean {
    const type = this.custom_type || this.subtype;
    return type === 'image' || type === 'batch';
  }

  hasThumbnails(): boolean {
    return Array.isArray(this.thumbnails) && this.thumbnails.length === 0
      ? false
      : this.thumbnails
      ? true
      : false;
  }

  toPlainObject() {
    const plainEntity = super.toPlainObject();
    if (plainEntity.remind_object && plainEntity.remind_object.__list) {
      delete plainEntity.remind_object.__list;
    }

    return plainEntity;
  }

  /**
   * Child models
   */
  childModels(): any {
    return {
      ownerObj: UserModel,
      remind_object: ActivityModel,
    };
  }

  /**
   * Get the activity thumb source
   * {uri: 'http...'}
   * @param {string} size
   */
  getThumbSource(size: ThumbSize = 'medium') {
    // for gif use always the same size to take advantage of the cache (they are not resized)
    if (this.isGif()) {
      size = 'xlarge';
    }

    if (this.thumbnails && this.thumbnails[size]) {
      return { uri: this.thumbnails[size], headers: api.buildHeaders() };
    }

    // fallback to old behavior
    if (this.paywall || this.paywall_unlocked) {
      const guid = this.entity_guid === '' ? this.guid : this.entity_guid;
      const unlock = this.isOwner() ? '?unlock_paywall=1' : '';
      return {
        uri: MINDS_URI + 'fs/v1/thumbnail/' + guid + unlock,
        headers: api.buildHeaders(),
      };
    }
    if (this.custom_type === 'batch') {
      return {
        uri: MINDS_CDN_URI + 'fs/v1/thumbnail/' + this.entity_guid + '/' + size,
        headers: api.buildHeaders(),
      };
    }
    return {
      uri: MINDS_CDN_URI + 'fs/v1/thumbnail/' + this.guid + '/' + size,
      headers: api.buildHeaders(),
    };
  }

  isGif(): boolean {
    return this.custom_data && this.custom_data[0] && this.custom_data[0].gif;
  }

  /**
   * Preload thumb on image cache
   */
  preloadThumb(size: ThumbSize = 'medium') {
    FastImage.preload([this.getThumbSource(size)]);
  }

  shouldBeBlured(): boolean {
    const user = sessionService.getUser();

    if (user && user.mature) {
      return false;
    }

    if (this.nsfw !== undefined) {
      let res = [1, 2, 4].filter((nsfw) => this.nsfw!.indexOf(nsfw) > -1)
        .length;
      if (res) {
        return true;
      }
    }

    if (typeof this.flags !== 'undefined') {
      return !!this.flags.mature;
    }

    if (typeof this.mature !== 'undefined') {
      return !!this.mature;
    }

    if (
      typeof this.custom_data !== 'undefined' &&
      typeof this.custom_data[0] !== 'undefined'
    ) {
      return !!this.custom_data[0].mature;
    }

    if (this.custom_data !== undefined) {
      return !!this.custom_data.mature;
    }

    return false;
  }

  /**
   * Get activity text
   */
  get text() {
    return this.message || this.description || '';
  }

  @action
  toggleMatureVisibility = () => {
    if (GOOGLE_PLAY_STORE || Platform.OS === 'ios') {
      showNotification(i18n.t('activity.notSafeComment'));
      return;
    }
    this.mature_visibility = !this.mature_visibility;

    if (this.remind_object && this.remind_object) {
      this.remind_object.mature_visibility = this.mature_visibility;
    }
  };

  @action
  setVisible(value: boolean) {
    this.is_visible = value;
    if (this.remind_object) {
      this.remind_object.is_visible = value;
    }
  }

  /**
   * Increment the comments counter
   */
  @action
  incrementCommentsCounter() {
    this['comments:count']++;
  }

  /**
   * Decrement the comments counter
   */
  @action
  decrementCommentsCounter() {
    this['comments:count']--;
  }

  /**
   * Unlock the activity and update data on success
   */
  @action
  async unlock(ignoreError = false) {
    try {
      const result: object | false = await wireService.unlock(this.guid);
      if (result) {
        // all changes should be atomic (trigger render only once)
        runInAction(() => {
          // create a new model because we need the child models
          const list = this.__list;
          const model: ActivityModel = ActivityModel.create(result);
          Object.assign(this, model);
          this.__list = list;
        });
      }

      if (
        this.entity_guid &&
        this.perma_url &&
        this.perma_url?.startsWith(MINDS_URI)
      ) {
        NavigationService.push('BlogView', {
          guid: this.entity_guid,
          unlock: true,
        });
      }

      return result;
    } catch (err) {
      if (!ignoreError) {
        Alert.alert(err.message);
      }
      return false;
    }
  }

  /**
   * Get the lock type for the activity
   */
  getLockType = (): LockType | undefined => {
    const support_tier: SupportTiersType | null =
      this.wire_threshold && 'support_tier' in this.wire_threshold
        ? this.wire_threshold.support_tier
        : null;
    if (!support_tier) {
      return;
    }
    let type: LockType = support_tier.public ? 'members' : 'paywall';

    if (mindsService.settings.plus.support_tier_urn === support_tier.urn) {
      type = 'plus';
    }

    return type;
  };

  /**
   * Unlock the entity or prompt pay options
   */
  async unlockOrPay() {
    const result = await this.unlock(true);

    if (result) {
      return;
    }

    const lockType = this.getLockType();

    const support_tier: SupportTiersType | null =
      this.wire_threshold && 'support_tier' in this.wire_threshold
        ? this.wire_threshold.support_tier
        : null;

    switch (lockType) {
      case 'plus':
        NavigationService.push('UpgradeScreen', {
          support_tier,
          entity: this,
          onComplete: (resultComplete: any) => {
            if (resultComplete && resultComplete.payload.method === 'onchain') {
              setTimeout(() => {
                Alert.alert(
                  i18n.t('wire.weHaveReceivedYourTransaction'),
                  i18n.t('wire.pleaseTryUnlockingMessage'),
                );
              }, 400);
            } else {
              this.unlock();
            }
          },
        });
        break;
      case 'members':
      case 'paywall':
        NavigationService.push('JoinMembershipScreen', {
          entity: this,
          onComplete: (resultComplete: any) => {
            if (resultComplete && resultComplete.payload.method === 'onchain') {
              setTimeout(() => {
                Alert.alert(
                  i18n.t('wire.weHaveReceivedYourTransaction'),
                  i18n.t('wire.pleaseTryUnlockingMessage'),
                );
              }, 400);
            } else {
              this.unlock();
            }
          },
        });
    }
  }

  @action
  async togglePin() {
    // allow owners only
    if (!this.isOwner()) {
      return;
    }

    try {
      this.pinned = !this.pinned;
      await setPinPost(this.guid, this.pinned);
    } catch (e) {
      this.pinned = !this.pinned;
      Alert.alert(i18n.t('errorPinnedPost'));
    }
  }

  @action
  async deleteEntity() {
    try {
      await deleteItem(this.guid);
      this.removeFromList();
      ActivityModel.events.emit('deleteEntity', this);
    } catch (err) {
      logService.exception('[ActivityModel]', err);
      throw err;
    }
  }

  async deleteRemind() {
    try {
      await api.delete(`api/v3/newsfeed/${this.urn}`);
      this.removeFromList();
      ActivityModel.events.emit('deleteEntity', this);
    } catch (err) {
      logService.exception('[ActivityModel]', err);
      throw err;
    }
  }

  @action
  async toggleFollow() {
    const method = this['is:following'] ? unfollow : follow;
    try {
      await method(this.guid);
      runInAction(() => {
        this['is:following'] = !this['is:following'];
      });
    } catch (err) {
      logService.exception('[OffsetFeedListStore]', err);
      throw err;
    }
  }

  @action
  setEdited(value) {
    this.edited = value;
  }
}

/**
 * Define model observables
 */
decorate(ActivityModel, {
  //@ts-ignore
  'thumbs:down:count': observable,
  'thumbs:up:count': observable,
  'comments:count': observable,
  'thumbs:down:user_guids': observable,
  'thumbs:up:user_guids': observable,
});
