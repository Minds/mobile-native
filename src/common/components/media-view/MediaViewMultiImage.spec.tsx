import { render, screen } from '@testing-library/react-native';
import * as React from 'react';
import ActivityModel from '../../../newsfeed/ActivityModel';
import MediaViewMultiImage from './MediaViewMultiImage';

describe('MediaViewMultiImage', () => {
  it('should render 2 images correctly', async () => {
    const entity = generateFakeMultiImageEntity({ images: 2 });
    render(
      <MediaViewMultiImage
        entity={entity}
        onImagePress={jest.fn()}
        onImageLongPress={jest.fn()}
      />,
    );
    checkImagesMatch(entity, 2);
    expect(screen.toJSON()).toMatchSnapshot();
  });
  it('should render 3 images correctly', async () => {
    const entity = generateFakeMultiImageEntity({ images: 3 });
    render(
      <MediaViewMultiImage
        entity={entity}
        onImagePress={jest.fn()}
        onImageLongPress={jest.fn()}
      />,
    );
    checkImagesMatch(entity, 3);
    expect(screen.toJSON()).toMatchSnapshot();
  });
  it('should render 4 images correctly', async () => {
    const entity = generateFakeMultiImageEntity({ images: 4 });
    render(
      <MediaViewMultiImage
        entity={entity}
        onImagePress={jest.fn()}
        onImageLongPress={jest.fn()}
      />,
    );
    checkImagesMatch(entity, 4);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});

const checkImagesMatch = (entity, length) => {
  let images: any[] = [];
  entity.custom_data.map(image => {
    const i = screen.getByTestId('image-' + image.src);
    if (i) {
      images.push(i);
    }
  });

  expect(images.length).toBe(length);
};

const generateFakeMultiImageEntity = (
  { images, blurred = false, locked = false } = { images: 2 },
) => {
  return {
    shouldBeBlured: () => blurred,
    isLocked: () => locked,
    custom_data: new Array(images)
      .fill({
        src: 'http://fake.com',
      })
      .map((image, index) => ({
        ...image,
        src: image.src + index,
      })),
  } as ActivityModel;
};
