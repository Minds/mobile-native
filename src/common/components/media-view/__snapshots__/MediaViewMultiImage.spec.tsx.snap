// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaViewMultiImage should render 2 images correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "height": 220,
      "marginHorizontal": 12,
      "marginTop": 12,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "column",
        "marginRight": 4,
      }
    }
  >
    <View
      activeOpacity={1}
      onLongPress={[Function]}
      onPress={[Function]}
      style={
        {
          "borderRadius": 3,
          "elevation": 2,
          "flex": 1,
          "height": "100%",
          "shadowColor": "rgba(0, 0, 0, 0.4)",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.4,
          "shadowRadius": 0,
        }
      }
      testID="image-http://fake.com0"
    >
      <View
        style={
          {
            "borderRadius": 3,
            "elevation": 2,
            "flex": 1,
            "height": "100%",
            "shadowColor": "rgba(0, 0, 0, 0.4)",
            "shadowOffset": {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 0,
          }
        }
      >
        <Image
          blurred={false}
          contentFit="cover"
          locked={false}
          onError={[Function]}
          onLoadEnd={[Function]}
          onProgress={[Function]}
          source={
            {
              "headers": {
                "App-Version": "3.8.0",
                "Cache-Control": "no-cache, no-store, must-revalidate",
                "Content-Type": "application/json",
                "Pragma": "no-cache",
                "no-cache": "1",
              },
              "uri": "http://fake.com0",
            }
          }
          style={
            {
              "borderRadius": 3,
              "elevation": 2,
              "flex": 1,
              "height": "100%",
              "shadowColor": "rgba(0, 0, 0, 0.4)",
              "shadowOffset": {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.4,
              "shadowRadius": 0,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "column",
        "marginLeft": 4,
      }
    }
  >
    <View
      activeOpacity={1}
      onLongPress={[Function]}
      onPress={[Function]}
      style={
        {
          "borderRadius": 3,
          "elevation": 2,
          "flex": 1,
          "height": "100%",
          "shadowColor": "rgba(0, 0, 0, 0.4)",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.4,
          "shadowRadius": 0,
        }
      }
      testID="image-http://fake.com1"
    >
      <View
        style={
          {
            "borderRadius": 3,
            "elevation": 2,
            "flex": 1,
            "height": "100%",
            "shadowColor": "rgba(0, 0, 0, 0.4)",
            "shadowOffset": {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 0,
          }
        }
      >
        <Image
          blurred={false}
          contentFit="cover"
          locked={false}
          onError={[Function]}
          onLoadEnd={[Function]}
          onProgress={[Function]}
          source={
            {
              "headers": {
                "App-Version": "3.8.0",
                "Cache-Control": "no-cache, no-store, must-revalidate",
                "Content-Type": "application/json",
                "Pragma": "no-cache",
                "no-cache": "1",
              },
              "uri": "http://fake.com1",
            }
          }
          style={
            {
              "borderRadius": 3,
              "elevation": 2,
              "flex": 1,
              "height": "100%",
              "shadowColor": "rgba(0, 0, 0, 0.4)",
              "shadowOffset": {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.4,
              "shadowRadius": 0,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`MediaViewMultiImage should render 3 images correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "height": 220,
      "marginHorizontal": 12,
      "marginTop": 12,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "column",
        "marginRight": 4,
      }
    }
  >
    <View
      activeOpacity={1}
      onLongPress={[Function]}
      onPress={[Function]}
      style={
        {
          "borderRadius": 3,
          "elevation": 2,
          "flex": 1,
          "height": "100%",
          "shadowColor": "rgba(0, 0, 0, 0.4)",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.4,
          "shadowRadius": 0,
        }
      }
      testID="image-http://fake.com0"
    >
      <View
        style={
          {
            "borderRadius": 3,
            "elevation": 2,
            "flex": 1,
            "height": "100%",
            "shadowColor": "rgba(0, 0, 0, 0.4)",
            "shadowOffset": {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 0,
          }
        }
      >
        <Image
          blurred={false}
          contentFit="cover"
          locked={false}
          onError={[Function]}
          onLoadEnd={[Function]}
          onProgress={[Function]}
          source={
            {
              "headers": {
                "App-Version": "3.8.0",
                "Cache-Control": "no-cache, no-store, must-revalidate",
                "Content-Type": "application/json",
                "Pragma": "no-cache",
                "no-cache": "1",
              },
              "uri": "http://fake.com0",
            }
          }
          style={
            {
              "borderRadius": 3,
              "elevation": 2,
              "flex": 1,
              "height": "100%",
              "shadowColor": "rgba(0, 0, 0, 0.4)",
              "shadowOffset": {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.4,
              "shadowRadius": 0,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "column",
        "marginLeft": 4,
      }
    }
  >
    <View
      activeOpacity={1}
      onLongPress={[Function]}
      onPress={[Function]}
      style={
        {
          "borderRadius": 3,
          "elevation": 2,
          "flex": 1,
          "height": "100%",
          "shadowColor": "rgba(0, 0, 0, 0.4)",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.4,
          "shadowRadius": 0,
        }
      }
      testID="image-http://fake.com1"
    >
      <View
        style={
          {
            "borderRadius": 3,
            "elevation": 2,
            "flex": 1,
            "height": "100%",
            "shadowColor": "rgba(0, 0, 0, 0.4)",
            "shadowOffset": {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 0,
          }
        }
      >
        <Image
          blurred={false}
          contentFit="cover"
          locked={false}
          onError={[Function]}
          onLoadEnd={[Function]}
          onProgress={[Function]}
          source={
            {
              "headers": {
                "App-Version": "3.8.0",
                "Cache-Control": "no-cache, no-store, must-revalidate",
                "Content-Type": "application/json",
                "Pragma": "no-cache",
                "no-cache": "1",
              },
              "uri": "http://fake.com1",
            }
          }
          style={
            {
              "borderRadius": 3,
              "elevation": 2,
              "flex": 1,
              "height": "100%",
              "shadowColor": "rgba(0, 0, 0, 0.4)",
              "shadowOffset": {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.4,
              "shadowRadius": 0,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        {
          "marginTop": 8,
        }
      }
    />
    <View
      activeOpacity={1}
      onLongPress={[Function]}
      onPress={[Function]}
      style={
        {
          "borderRadius": 3,
          "elevation": 2,
          "flex": 1,
          "height": "100%",
          "shadowColor": "rgba(0, 0, 0, 0.4)",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.4,
          "shadowRadius": 0,
        }
      }
      testID="image-http://fake.com2"
    >
      <View
        style={
          {
            "borderRadius": 3,
            "elevation": 2,
            "flex": 1,
            "height": "100%",
            "shadowColor": "rgba(0, 0, 0, 0.4)",
            "shadowOffset": {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 0,
          }
        }
      >
        <Image
          blurred={false}
          contentFit="cover"
          locked={false}
          onError={[Function]}
          onLoadEnd={[Function]}
          onProgress={[Function]}
          source={
            {
              "headers": {
                "App-Version": "3.8.0",
                "Cache-Control": "no-cache, no-store, must-revalidate",
                "Content-Type": "application/json",
                "Pragma": "no-cache",
                "no-cache": "1",
              },
              "uri": "http://fake.com2",
            }
          }
          style={
            {
              "borderRadius": 3,
              "elevation": 2,
              "flex": 1,
              "height": "100%",
              "shadowColor": "rgba(0, 0, 0, 0.4)",
              "shadowOffset": {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.4,
              "shadowRadius": 0,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`MediaViewMultiImage should render 4 images correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "height": 220,
      "marginHorizontal": 12,
      "marginTop": 12,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "column",
        "marginRight": 4,
      }
    }
  >
    <View
      activeOpacity={1}
      onLongPress={[Function]}
      onPress={[Function]}
      style={
        {
          "borderRadius": 3,
          "elevation": 2,
          "flex": 1,
          "height": "100%",
          "shadowColor": "rgba(0, 0, 0, 0.4)",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.4,
          "shadowRadius": 0,
        }
      }
      testID="image-http://fake.com0"
    >
      <View
        style={
          {
            "borderRadius": 3,
            "elevation": 2,
            "flex": 1,
            "height": "100%",
            "shadowColor": "rgba(0, 0, 0, 0.4)",
            "shadowOffset": {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 0,
          }
        }
      >
        <Image
          blurred={false}
          contentFit="cover"
          locked={false}
          onError={[Function]}
          onLoadEnd={[Function]}
          onProgress={[Function]}
          source={
            {
              "headers": {
                "App-Version": "3.8.0",
                "Cache-Control": "no-cache, no-store, must-revalidate",
                "Content-Type": "application/json",
                "Pragma": "no-cache",
                "no-cache": "1",
              },
              "uri": "http://fake.com0",
            }
          }
          style={
            {
              "borderRadius": 3,
              "elevation": 2,
              "flex": 1,
              "height": "100%",
              "shadowColor": "rgba(0, 0, 0, 0.4)",
              "shadowOffset": {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.4,
              "shadowRadius": 0,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        {
          "marginTop": 8,
        }
      }
    />
    <View
      activeOpacity={1}
      onLongPress={[Function]}
      onPress={[Function]}
      style={
        {
          "borderRadius": 3,
          "elevation": 2,
          "flex": 1,
          "height": "100%",
          "shadowColor": "rgba(0, 0, 0, 0.4)",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.4,
          "shadowRadius": 0,
        }
      }
      testID="image-http://fake.com2"
    >
      <View
        style={
          {
            "borderRadius": 3,
            "elevation": 2,
            "flex": 1,
            "height": "100%",
            "shadowColor": "rgba(0, 0, 0, 0.4)",
            "shadowOffset": {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 0,
          }
        }
      >
        <Image
          blurred={false}
          contentFit="cover"
          locked={false}
          onError={[Function]}
          onLoadEnd={[Function]}
          onProgress={[Function]}
          source={
            {
              "headers": {
                "App-Version": "3.8.0",
                "Cache-Control": "no-cache, no-store, must-revalidate",
                "Content-Type": "application/json",
                "Pragma": "no-cache",
                "no-cache": "1",
              },
              "uri": "http://fake.com2",
            }
          }
          style={
            {
              "borderRadius": 3,
              "elevation": 2,
              "flex": 1,
              "height": "100%",
              "shadowColor": "rgba(0, 0, 0, 0.4)",
              "shadowOffset": {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.4,
              "shadowRadius": 0,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "column",
        "marginLeft": 4,
      }
    }
  >
    <View
      activeOpacity={1}
      onLongPress={[Function]}
      onPress={[Function]}
      style={
        {
          "borderRadius": 3,
          "elevation": 2,
          "flex": 1,
          "height": "100%",
          "shadowColor": "rgba(0, 0, 0, 0.4)",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.4,
          "shadowRadius": 0,
        }
      }
      testID="image-http://fake.com1"
    >
      <View
        style={
          {
            "borderRadius": 3,
            "elevation": 2,
            "flex": 1,
            "height": "100%",
            "shadowColor": "rgba(0, 0, 0, 0.4)",
            "shadowOffset": {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 0,
          }
        }
      >
        <Image
          blurred={false}
          contentFit="cover"
          locked={false}
          onError={[Function]}
          onLoadEnd={[Function]}
          onProgress={[Function]}
          source={
            {
              "headers": {
                "App-Version": "3.8.0",
                "Cache-Control": "no-cache, no-store, must-revalidate",
                "Content-Type": "application/json",
                "Pragma": "no-cache",
                "no-cache": "1",
              },
              "uri": "http://fake.com1",
            }
          }
          style={
            {
              "borderRadius": 3,
              "elevation": 2,
              "flex": 1,
              "height": "100%",
              "shadowColor": "rgba(0, 0, 0, 0.4)",
              "shadowOffset": {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.4,
              "shadowRadius": 0,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        {
          "marginTop": 8,
        }
      }
    />
    <View
      activeOpacity={1}
      onLongPress={[Function]}
      onPress={[Function]}
      style={
        {
          "borderRadius": 3,
          "elevation": 2,
          "flex": 1,
          "height": "100%",
          "shadowColor": "rgba(0, 0, 0, 0.4)",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.4,
          "shadowRadius": 0,
        }
      }
      testID="image-http://fake.com3"
    >
      <View
        style={
          {
            "borderRadius": 3,
            "elevation": 2,
            "flex": 1,
            "height": "100%",
            "shadowColor": "rgba(0, 0, 0, 0.4)",
            "shadowOffset": {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 0,
          }
        }
      >
        <Image
          blurred={false}
          contentFit="cover"
          locked={false}
          onError={[Function]}
          onLoadEnd={[Function]}
          onProgress={[Function]}
          source={
            {
              "headers": {
                "App-Version": "3.8.0",
                "Cache-Control": "no-cache, no-store, must-revalidate",
                "Content-Type": "application/json",
                "Pragma": "no-cache",
                "no-cache": "1",
              },
              "uri": "http://fake.com3",
            }
          }
          style={
            {
              "borderRadius": 3,
              "elevation": 2,
              "flex": 1,
              "height": "100%",
              "shadowColor": "rgba(0, 0, 0, 0.4)",
              "shadowOffset": {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.4,
              "shadowRadius": 0,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
